### DISCUSSION TITLE
Should the game immediately clear the fog of war when the Hero's sout range is changed?

### DISCUSSION THREAD

**Discussion by Districh-ru** [ihhub/fheroes2#6545] [General]
Currently in fheroes2, as in original game, when Hero picks an artifact, gets it from another Hero, or when a Hero is freed from the Jail the game does not reveal the fog of war in his scout range immediately after this action.
The fog of war is revealed only with the next move of the Hero, who captured the artifact or with the move of the Hero rescued from Jail.
But learning a Scout skill immediately reveals the fog of war in the new scout area.

Should we:
1. Keep it, as it matches the original game.
2. Disable immediate reveal after learning the Scout Skill.
3. Reveal the area in Hero scout range after any change of scout range immediately after this change.

**Comment by oleg-derevenetz** [comment]
Hi @LeHerosInconnu you are a game logic expert, what do you think?

**Reply by LeHerosInconnu** [reply]
Hello @oleg-derevenetz and @Districh-ru,

> Hi @LeHerosInconnu you are a game logic expert, what do you think?

Always update the hero's vision radius as soon as there is a change (proposal 3) seems to me the most logical, if only to keep an uniformity in the game.

**Reply by zenseii** [reply]
I support this. If you are now better at scouting, there shouldn't be a mandatory single step to take before you can actually see further.

**Reply by Districh-ru** [reply]
Hello @LeHerosInconnu, @zenseii, @oleg-derevenetz.
While fixing issues in PR #6514 I've already made the changes written in proposal 3.
But then I've checked the original game and returned to proposal 1.
I've kept the code for proposal 3 and if the team will decide to implement it I can do it in PR #6514. Or (maybe better) open a new PR after #6514 is merged.

**Reply by oleg-derevenetz** [reply]
@Districh-ru I think this could be implemented in #6514, no need to create a separate PR.

**Reply by Districh-ru** [reply]
@oleg-derevenetz, I've talked to @ihhub and @Branikolog in Discord earlier and they agreed with proposal 3, so I committed it.
If there will be objections we can revert these changes.
